core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -73,61 +72,43 @@ public class DefaultBlockWorkerClient implements BlockWorkerClient {
   private final GrpcServerAddress mAddress;
   private final long mRpcTimeoutMs;
 
-  private final BlockWorkerGrpc.BlockWorkerStub mStreamingAsyncStub;
-  private final BlockWorkerGrpc.BlockWorkerBlockingStub mRpcBlockingStub;
-  private final BlockWorkerGrpc.BlockWorkerStub mRpcAsyncStub;
+  private BlockWorkerGrpc.BlockWorkerStub mStreamingAsyncStub;
+  private BlockWorkerGrpc.BlockWorkerBlockingStub mRpcBlockingStub;
+  private BlockWorkerGrpc.BlockWorkerStub mRpcAsyncStub;
 
   @Nullable
   private final ResourceLeakTracker<DefaultBlockWorkerClient> mTracker;
 
   /**
    * Creates a client instance for communicating with block worker.
    *
-   * @param userState     the user state
+   * @param context the client context
    * @param address     the address of the worker
    * @param alluxioConf Alluxio configuration
    */
-  public DefaultBlockWorkerClient(UserState userState, GrpcServerAddress address,
-      AlluxioConfiguration alluxioConf) throws IOException {
-    RetryPolicy retryPolicy = RetryUtils.defaultClientRetry();
-    UnauthenticatedException lastException = null;
-    // TODO(feng): unify worker client with AbstractClient
-    while (retryPolicy.attempt()) {
-      try {
-        // Disables channel pooling for data streaming to achieve better throughput.
-        // Channel is still reused due to client pooling.
-        mStreamingChannel = GrpcChannelBuilder.newBuilder(address, alluxioConf)
-            .setSubject(userState.getSubject())
-            .setNetworkGroup(GrpcNetworkGroup.STREAMING)
-            .build();
-        mStreamingChannel.intercept(new StreamSerializationClientInterceptor());
-        // Uses default pooling strategy for RPC calls for better scalability.
-        mRpcChannel = GrpcChannelBuilder.newBuilder(address, alluxioConf)
-            .setSubject(userState.getSubject())
-            .setNetworkGroup(GrpcNetworkGroup.RPC)
-            .build();
-        lastException = null;
-        break;
-      } catch (StatusRuntimeException e) {
-        close();
-        throw AlluxioStatusException.fromStatusRuntimeException(e);
-      } catch (UnauthenticatedException e) {
-        close();
-        userState.relogin();
-        lastException = e;
-      }
-    }
-    if (lastException != null) {
-      throw lastException;
-    }
-    mStreamingAsyncStub = BlockWorkerGrpc.newStub(mStreamingChannel);
-    mRpcBlockingStub = BlockWorkerGrpc.newBlockingStub(mRpcChannel);
-    mRpcAsyncStub = BlockWorkerGrpc.newStub(mRpcChannel);
+  public DefaultBlockWorkerClient(ClientContext context, GrpcServerAddress address,
+                                  AlluxioConfiguration alluxioConf) {
+    // we don't use InetSocketAddress
+    // we use default retry policy for connection and no retry for actual rpc
+    super(context, null, RetryUtils::defaultClientRetry, RetryUtils::oneshotRetryPolicy);
```

Comment:
As commented, we retry normally for connection but don't retry for real rpc calls.